src/serialport.cpp

```diff
@@ -87,75 +87,74 @@ static void deleteQForFD(const int fd) {
 
 
 NAN_METHOD(Open) {
-  NanScope();
 
   // path
-  if(!args[0]->IsString()) {
-    NanThrowTypeError("First argument must be a string");
-    NanReturnUndefined();
+  if(!info[0]->IsString()) {
+    Nan::ThrowTypeError("First argument must be a string");
+    return;
   }
-  v8::String::Utf8Value path(args[0]->ToString());
+  v8::String::Utf8Value path(info[0]->ToString());
 
   // options
-  if(!args[1]->IsObject()) {
-    NanThrowTypeError("Second argument must be an object");
-    NanReturnUndefined();
+  if(!info[1]->IsObject()) {
+    Nan::ThrowTypeError("Second argument must be an object");
+    return;
   }
-  v8::Local<v8::Object> options = args[1]->ToObject();
+  v8::Local<v8::Object> options = info[1]->ToObject();
 
   // callback
-  if(!args[2]->IsFunction()) {
-    NanThrowTypeError("Third argument must be a function");
-    NanReturnUndefined();
+  if(!info[2]->IsFunction()) {
+    Nan::ThrowTypeError("Third argument must be a function");
+    return;
   }
-  v8::Local<v8::Function> callback = args[2].As<v8::Function>();
+  v8::Local<v8::Function> callback = info[2].As<v8::Function>();
 
   OpenBaton* baton = new OpenBaton();
   memset(baton, 0, sizeof(OpenBaton));
   strcpy(baton->path, *path);
-  baton->baudRate = options->Get(NanNew<v8::String>("baudRate"))->ToInt32()->Int32Value();
-  baton->dataBits = options->Get(NanNew<v8::String>("dataBits"))->ToInt32()->Int32Value();
-  baton->bufferSize = options->Get(NanNew<v8::String>("bufferSize"))->ToInt32()->Int32Value();
-  baton->parity = ToParityEnum(options->Get(NanNew<v8::String>("parity"))->ToString());
-  baton->stopBits = ToStopBitEnum(options->Get(NanNew<v8::String>("stopBits"))->ToNumber()->NumberValue());
-  baton->rtscts = options->Get(NanNew<v8::String>("rtscts"))->ToBoolean()->BooleanValue();
-  baton->xon = options->Get(NanNew<v8::String>("xon"))->ToBoolean()->BooleanValue();
-  baton->xoff = options->Get(NanNew<v8::String>("xoff"))->ToBoolean()->BooleanValue();
-  baton->xany = options->Get(NanNew<v8::String>("xany"))->ToBoolean()->BooleanValue();
-  baton->hupcl = options->Get(NanNew<v8::String>("hupcl"))->ToBoolean()->BooleanValue();
-
-  v8::Local<v8::Object> platformOptions = options->Get(NanNew<v8::String>("platformOptions"))->ToObject();
+  baton->baudRate = Nan::Get(options, Nan::New<v8::String>("baudRate").ToLocalChecked()).ToLocalChecked()->ToInt32()->Int32Value();
```

Comment:
@ghostoy talking about this double tolocalchecked. Are we sure it needs to be checked twice?